### PR TITLE
Added the sample demonstrating Index Management imitating the .NET sample

### PR DESCRIPTION
**Pull request by HussainMir**
Hi,

Request you to check. Just wondering about which style of addressing to stick with or is preferred, i.e. _self based or name / id based.

Thanks.

### CODE REVIEW

**Comment by moderakh** [comment]
Hussain,


name base link is preferred. Please use name base link not self link in the example.


As self link may confuse the end user.


Thanks


Sent from Outlook<http://aka.ms/weboutlook>

________________________________

File: samples/DocumentManagement/Program.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+HOST = cfg.settings['host']
+MASTER_KEY = cfg.settings['master_key']
+DATABASE_ID = cfg.settings['database_id']
+COLLECTION_ID = cfg.settings['collection_id']
+
+database_link = 'dbs/' + DATABASE_ID
+collection_link = database_link + '/colls/' + COLLECTION_ID
+
+class IDisposable:
```
[the remark this answers is not in the document]
Comment (HussainMir) [reply samples/DocumentManagement/Program.py]: This is was present before. I removed this in Index Management project. Shall we remove this from other projects as well? I simply used helper methods instead.

File: samples/DocumentManagement/Program.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
+import pydocumentdb.documents as documents
```
[the remark this answers is not in the document]
Comment (HussainMir) [reply samples/DocumentManagement/Program.py]: Tested to be working for Python 2.7

File: python_vs2017.pyproj [hunk 3]
```diff
@@ -0,0 +1,95 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (HussainMir) [reply python_vs2017.pyproj]: Glad to see that, but some people are having difficulty to find that constants are being picked from a separate (shared) file config.py

File: samples/IndexManagement/Program.py [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+# Setup the certificate file in .pem format. 
+# If you still get an SSLError, try disabling certificate verification and suppress warnings
+
+def ObtainClient():
+    connection_policy = documents.ConnectionPolicy()
+    connection_policy.SSLConfiguration = documents.SSLConfiguration()
+    # Try to setup the cacert.pem
+    # connection_policy.SSLConfiguration.SSLCaCerts = CaCertPath
+    # Else, disable verification
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```
[the remark this answers is not in the document]
Comment (HussainMir) [reply samples/IndexManagement/Program.py]: To be honest, I did not use emulator and I was running live on my cosmos db account, and I was having trouble setting up the cacert.pem. Hence, in order to proceed, I did this.

File: samples/IndexManagement/Program.py [hunk 5]
```diff
@@ -0,0 +1,657 @@
...
+            pass
+        else: 
+            raise errors.HTTPFailure(e.status_code)
+
+def DeleteCollectionIfExists(client, database_link, collection_id):
+    try:
+        collection_link = "dbs/{0}".format(DATABASE_ID) + "/" + 'colls/{0}'.format(collection_id)
+        #collection_link =  database_link +  'colls/{0}'.format(collection_id) 
+        # having some trouble with this format of querying, raised an issue
+        # https://github.com/Azure/azure-documentdb-python/issues/106
```
[the remark this answers is not in the document]
Comment (HussainMir) [reply samples/IndexManagement/Program.py]: Removed. And used name based addressing throughout.

File: samples/IndexManagement/Program.py [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+        # Insert some documents
+        collection_link = GetCollectionLink(database_id, COLLECTION_ID)
+        doc1 = client.CreateDocument(collection_link, { "id" : "dyn1", "length" : 10, "width" : 5, "height" : 15 })
+        doc2 = client.CreateDocument(collection_link, { "id" : "dyn2", "length" : 7, "width" : 15 })
+        doc3 = client.CreateDocument(collection_link, { "id" : "dyn3", "length" : 2 })
+        print("Three docs created with ids : ", doc1["id"], doc2["id"], doc3["id"])
+
+        # Switch to lazy indexing and wait till complete.
+        print("Chaning the indexing mode of the collection from default to lazy")
+        created_collection['indexingPolicy']['indexingMode'] = documents.IndexingMode.Lazy
```
Comment (moderakh) [inline samples/IndexManagement/Program.py]: please do not add lazy indexing sample. (remove)

Comment (HussainMir) [reply samples/IndexManagement/Program.py]: Removed this part which was demonstrating the use of Lazy indexing. Now I only have one line in the beginning of the file (description) "*indexingMode can be either of consistent, lazy or none*"

**Pull request merged**